Makefile

```diff
@@ -1,7 +1,8 @@
 TARGET = ciadpi
 
-#CPPFLAGS = -D_XOPEN_SOURCE=500
+CPPFLAGS = -D_XOPEN_SOURCE=600
```

Comment:
> with _XOPEN_SOURCE _BSD_VISIBLE will not be defined

If you about FreeBSD, __BSD_VISIBLE don't get defined when _POSIX_SOURCE, _POSIX_C_SOURCE, _ANSI_SOURCE, _C99_SOURCE or _C11_SOURCE are defined:
https://svnweb.freebsd.org/base/head/sys/sys/cdefs.h?view=markup#l755

_XOPEN_SOURCE=600 defines POSIX_C_SOURCE=200112, this is why it don't get defined. I think not defining any POSIX_C_SOURCE with posix compatibility is not a good idea. 

Also _DEFAULT_SOURCE is not used by freebsd by this link so this is why __BSD_VISIBLE gets defined, but with glibc _DEFAULT_SOURCE defines _POSIX_C_SOURCE=200809L:
https://sourceware.org/git/?p=glibc.git;a=blob;f=include/features.h#l300
Honestly I don't know if this intended or compatibility coincidence.

Although _DEFAULT_SOURCE on Linux looks like POSIX with BSD extensions, but from `man feature_test_macros`:
> Defining _DEFAULT_SOURCE without defining other  individual  macros  or invoking the compiler in one of its "standard" modes has no effect.

Reply:
> Defining _DEFAULT_SOURCE without defining other individual macros or invoking the compiler in one of its "standard" modes has no effect.

it is needed because `-std=c99` was declared.

Reply:
The problem also that BSD's libc don't use _DEFAULT_SOURCE and will default to their own implementations compatibility when _XOPEN_SOURCE=600 at least sets POSIX standard and most unix libc supports it. Although you can't set POSIX on freebsd without breaksing __BSD_VISIBLE :(

Reply:
maybe it's worth removing std=c99 and -D_XOPEN_SOURCE?

Reply:
Well, if we define _DEFAULT_SOURCE:
1. On Linux with glibc it will set POSIX.1-2008, BSD and System V extensions.
2. On FreeBSD it will be ignored, thus by default it will set POSIX.1-2008, X/Open 700 and BSD (__BSD_VISIBLE) extensions
By mailing lists FreeBSD deliberately enables __BSD_VISIBLE only when nothing is defined because "Standards definitions should restrict features, not add them":
https://lists.freebsd.org/pipermail/freebsd-standards/2004-March/000485.html

3. On NetBSD it will be ignored, thus by default it will set _NETBSD_SOURCE
https://anonhg.netbsd.org/src/file/tip/sys/sys/featuretest.h
4. On OpenBSD it will be ignored, thus by default it will set POSIX.1-2008, X/Open 700 and BSD (__BSD_VISIBLE) extensions
https://cvsweb.openbsd.org/src/sys/sys/cdefs.h?rev=1.43&content-type=text/x-cvsweb-markup&only_with_tag=MAIN

SO if you are fine with POSIX with BSD extensions on most systems then is can be used.

> std=c99

This is another problem btw, currently the code actually is not fully c99 compliant because at least it uses anonymous structs fields which is GNU extension (And defined in C standard only in C11), but gnu extensions for c standards is another problem :). You can check this by adding `-pedantic` to CFLAGS so it will fully force specified c standard.